src/DotVVM.Framework.Hosting.AspNetCore/ServiceCollectionExtensions.cs

```diff
@@ -11,11 +11,15 @@
 using DotVVM.Framework.Compilation;
 using DotVVM.Framework.Runtime;
 using DotVVM.Framework.Hosting.AspNetCore;
+using DotVVM.Framework.Diagnostics;
+using DotVVM.Framework.Runtime.Tracing;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
     public static class ServiceCollectionExtensions
     {
+        private static readonly DiagnosticsStartupTracer startupTracer = new DiagnosticsStartupTracer();
```

Comment:
It is added to the `IServiceCollection` container later (at the end of the file). 
In ASP.NET Core, it needs to be a static field because we need it before the `IServiceCollection` is ready and because it is called from both `ConfigureServices` and `Configure` methods in `Startup`.
In OWIN it can be just a local variable as there is only one method for initialization of DotVVM.